pkg/http/middlewares.go

```diff
@@ -43,8 +42,12 @@ func APITokenAuthMiddleware(token string) func(next http.Handler) http.Handler {
 		}
 
 		return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+			// Check if the route is excluded from API token auth
+			endpointData, _ := r.Context().Value(endpoints.EndpointCtxKey{}).(*endpoints.EndpointCtxData)
+			excluded := endpointData != nil && endpointData.Settings.BypassAPITokenAuth
+
 			v := r.Header.Get(authConsts.APITokenHeader)
-			if v != token && !isRouteExcludedFromAPITokenAuth(r.Method, r.URL) {
+			if v != token && !excluded {
```

Comment:
This method caused a CVE. It's been changed now so the route itself defines if it can be excluded by API token auth